src/compile/repeater.ts

```diff
@@ -57,32 +57,50 @@ function replaceRepeaterInFieldDef(fieldDef: ScaleFieldDef<Field>, repeater: Rep
   return fieldDef as ScaleFieldDef<string>;
 }
 
+function replaceRepeaterInChannelDef(channelDef: ChannelDef<Field>, repeater: RepeaterValue): ChannelDef<string> {
+  if (isConditionalDef(channelDef) && isFieldDef(channelDef.condition)) {
+    const fd = replaceRepeaterInFieldDef(channelDef.condition, repeater);
+    if (fd) {
+      return {
+        ...channelDef,
+        condition: fd
+      } as ChannelDef<string>;
+    } else {
+      return {
+        value: (channelDef as ValueDef).value
+      };
+    }
+  } else if (isFieldDef(channelDef)) {
+    const fd = replaceRepeaterInFieldDef(channelDef, repeater);
+    if (fd) {
+      return fd;
+    } else if (isConditionalDef(channelDef)) {
+      return {value: channelDef.condition.value};
+    } else {
+      return undefined;
+    }
+  } else {
+    return channelDef as ValueDef;
```

Comment:
I think you do not need `as` if you organize the `if/else` like this:

```
if (isFieldDef(channelDef)) {
   ...
} else { // Definitely ValueDef
  if (isConditionalDef(channelDef) && isFieldDef(channelDef.condition)) {
    ...
  }
  return channelDef;
}

Reply:
That would add additional runtime checks, wouldn't it?

Reply:
How?  I think in general, try to separate a branch for a field def and a value def makes it easier to read.
The currently logic here as value def in two separate branches.

Reply:
I see. Will update.

Reply:
https://github.com/vega/vega-lite/pull/3006/commits/1815cec3f3d4dce93251fc2fbe5e6be183222dc1